### PR TITLE
Initial Context page

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
This PR is a first implementation that loads a given context loading plugin configuration from a remote server. 

Page at path `context/<contextId>/<mapId>` will load the context loading. A sample context is loaded on resource `11516` on dev.mapstore.geo-solutions.it.
example: 
`http://localhost:8081/?debug=true#/context/11516/10279`

Using `context/default/new` will load plugins from localConfig and the map `new`

## Issues
 - #4380 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Feature
 
**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

--> THE PR is in draft because still need to handle some issues:

 - [ ]  feedback mask works but login is not shown when context is not accessible.
 - [ ] need to add translations
 - [ ] some missing tests for context selector. 

But you can give a look and/or try

--> Had to convert mapConfig thunk into observable to make it work.

### CODE REVIEW

**Comment by mbarto** [comment]
Please remove npm-debug.log.2933983051 from commit

File: web/client/actions/__tests__/config-test.js [hunk 1]
```diff
@@ -7,50 +7,32 @@
...
-        loadMapConfig('missingConfig.json')((e) => {
-            try {
-                expect(e).toExist();
-                expect(e.type).toBe(MAP_CONFIG_LOAD_ERROR);
-                done();
-            } catch (ex) {
-                done(ex);
-            }
-        });
+    it('loadMapConfig', () => {
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/actions/__tests__/config-test.js]: These tests has been moved in the epic, if you look at that, the descriptions are the same strngs. 
So now here we have only actionCreator test.

File: web/client/actions/config.js [hunk 2]
```diff
@@ -32,20 +33,10 @@ function configureError(e, mapId) {
...
-                try {
-                    JSON.parse(response.data);
-                } catch (e) {
-                    dispatch(configureError('Configuration file broken (' + configName + '): ' + e.message, mapId));
-                }
-            }
-        }).catch((e) => {
-            dispatch(configureError(e, mapId));
-        });
+    return {
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/actions/config.js]: Yes is needed, I undestand this is not immediate. 
Now this action have to be triggered by the epic of context loading. 
The epic  + thunk middleware configured actually in MapStore allow to return thunks. 
But in this case I need to intercept in the same epic some actions dispatched by this thunk (configureMap and configureErrror) immediately after triggering loadMapConfig. This was not working (probably a bug or a  limit / compatibility issue between the two middleware) Restructuring the thunk in an epic did the job

File: web/client/components/app/StandardApp.jsx [hunk 3]
```diff
@@ -122,9 +122,14 @@ class StandardApp extends React.Component {
         const {pluginsDef, appStore, initialActions, appComponent, mode, ...other} = this.props;
         const App = dragDropContext(html5Backend)(this.props.appComponent);
         return this.state.initialized ?
-            <ErrorBoundary><Provider store={this.store}>
-                <App {...other} plugins={assign(PluginsUtils.getPlugins(plugins), {requires})}/>
-            </Provider></ErrorBoundary>
+            <ErrorBoundary onError={(e) => {
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/components/app/StandardApp.jsx]: ok did it

File: web/client/epics/config.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import Rx from 'rxjs';
+import axios from '../libs/ajax';
+import {LOAD_MAP_CONFIG, configureMap, configureError} from '../actions/config';
+
+export const loadMapConfigAndConfigureMap = action$ =>
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/epics/config.js]: see the comment above

File: web/client/plugins/Context.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import * as epics from '../epics/context';
+
+import context from '../reducers/context';
+import { createPlugin } from '../utils/PluginsUtils';
+
+/**
+ * Plugin for GeoStory visualization
```
Comment (offtherailz) [inline web/client/plugins/Context.jsx]:
```suggestion
 * Plugin for Context support. Adds state handlers to the application (reducer and epics).
```

File: web/client/plugins/Context.jsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+ * @name Context
+ * @memberof plugins
+ */
+export default createPlugin("Context", {
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/plugins/Context.jsx]: Adds epic and reducer. For the moment nothing else. Should I add them to standard store instead?

[the remark this answers is not in the document]
Comment (mbarto) [reply web/client/plugins/Context.jsx]: yes, I think it's better

File: web/client/product/pages/Context.jsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        createStructuredSelector({
+            pluginsConfig: pluginsSelector,
+            mode: () => 'desktop',
+            context: currentContextSelector,
+            monitoredState: contextMonitoredStateSelector
+        }),
+        {
+            loadContext
+        }),
+    lifecycle({
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/product/pages/Context.jsx]: Contextually to this change (This and proptype) I added also documentation about how to configure this page.

File: web/client/product/pages/Context.jsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+import { contextMonitoredStateSelector, currentContextSelector, pluginsSelector } from '../../selectors/context';
+
+class Context extends React.Component {
+    static propTypes = {
+        mode: PropTypes.string,
+        match: PropTypes.object,
+        onInit: PropTypes.func,
+        plugins: PropTypes.object,
+        wrappedContainer: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+        location: PropTypes.object
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/product/pages/Context.jsx]: did it.

**Pull request merged**